src/containers/my-courses/add-course-with-input/AddCourseWithInput.tsx

```diff
@@ -11,16 +11,34 @@ import AppButton from '~/components/app-button/AppButton'
 import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
 import CoursesFilterBar from '~/containers/find-course/courses-filter-bar/CoursesFilterBar'
 import { styles } from '~/containers/my-courses/add-course-with-input/AddCourseWithInput.styles'
+import CoursesFiltersDrawer from '~/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer'
+import useForm from '~/hooks/use-form'
+import { useDrawer } from '~/hooks/use-drawer'
+import { CourseFilters } from '~/types'
 
 const AddCourseWithInput = () => {
   const [inputValue, setInputValue] = useState<string>('')
+  const { openDrawer, closeDrawer, isOpen } = useDrawer()
   const { t } = useTranslation()
 
+  const {
+    data: filters,
+    handleNonInputValueChange,
+    resetData
+  } = useForm<CourseFilters>({
+    initialValues: {
+      category: '',
+      subject: '',
+      proficiencyLevel: []
+    }
+  })
+
   const onChange = (e: ChangeEvent<HTMLInputElement>) => {
     setInputValue(e.target.value)
   }
-
   const onClear = () => setInputValue('')
+  const handleToggle = () => (isOpen ? closeDrawer() : openDrawer())
+  const handleClose = () => closeDrawer()
```

Comment:
I think, creation of this function  is redundant, you can just pass closeDrawer instead of handleClose